kdl_parser/src/kdl_parser.cpp

```diff
@@ -170,42 +176,50 @@ bool treeFromParam(const string& param, Tree& tree)
   return treeFromUrdfModel(robot_model, tree);
 }
 
-bool treeFromString(const string& xml, Tree& tree)
+bool treeFromString(const std::string & xml, KDL::Tree & tree)
 {
   TiXmlDocument urdf_xml;
   urdf_xml.Parse(xml.c_str());
   return treeFromXml(&urdf_xml, tree);
 }
 
-bool treeFromXml(TiXmlDocument *xml_doc, Tree& tree)
+bool treeFromXml(TiXmlDocument * xml_doc, KDL::Tree & tree)
 {
   urdf::Model robot_model;
-  if (!robot_model.initXml(xml_doc)){
+  if (!robot_model.initXml(xml_doc)) {
     ROS_ERROR("Could not generate robot model");
     return false;
   }
   return treeFromUrdfModel(robot_model, tree);
 }
 
 
-bool treeFromUrdfModel(const urdf::ModelInterface& robot_model, Tree& tree)
+bool treeFromUrdfModel(const urdf::ModelInterface & robot_model, KDL::Tree & tree)
 {
-  if (!robot_model.getRoot())
+  if (!robot_model.getRoot()) {
     return false;
+  }
 
-  tree = Tree(robot_model.getRoot()->name);
+  tree = KDL::Tree(robot_model.getRoot()->name);
 
   // warn if root link has inertia. KDL does not support this
-  if (robot_model.getRoot()->inertial)
-    ROS_WARN("The root link %s has an inertia specified in the URDF, but KDL does not support a root link with an inertia.  As a workaround, you can add an extra dummy link to your URDF.", robot_model.getRoot()->name.c_str());
+  if (robot_model.getRoot()->inertial) {
+    auto error_msg = std::string("The root link") +
```

Comment:
this is C++11 and cannot go upstream as is

Reply:
Yeah, good point, I fixed this.